LanguageFeatures/Static-access-shorthand/equality_A02_t01.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion For ==, we special-case when the right operand is (precisely!) a
+/// static member shorthand.
+/// ...
+/// This special-casing is only against an immediate static member shorthand. It
+/// does not change the context type of the second operand, so it would not work
+/// with, for example, `Endian.host == wantBig ? .big : .little`. Here the
+/// second operand is not a `<staticMemberShorthand>`, so it won't have a
+/// shorthand context set, and the context type of the second operand of `==` is
+/// the empty context `_`. (It's neither the static type of the first operand,
+/// nor the parameter type of the first operand's `operator==`.)
+///
+/// @description Checks that it is a compile-time error if the right operand of
+/// `==` operator isn not a precisely static member shorthand but a ternary
```

Comment:
There are a few occurrences of this one:
```suggestion
/// `==` operator is not a precisely static member shorthand but a ternary
```